File: Isomorphic_String.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
      //  Map <Character,Integer>map=new HashMap<>();
      int map1[]=new int[200];
      int map2[]=new int[200];
        for(int i=0;i<s.length();i++){
       
             System.out.println(map1[s.charAt(i)]+"->"+s.charAt(i)+"  "+map2[t.charAt(i)]+"->"+t.charAt(i));
          if(map1[s.charAt(i)]!=map2[t.charAt(i)])
                return false;

         map1[s.charAt(i)]=1+i;
         map2[t.charAt(i)]=1+i;
            System.out.println((map1[s.charAt(i)]=1+i));
            System.out.println((map2[t.charAt(i)]=1+i));
        }

      return true;
    }
}

File: python/String/isomorphic string.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:

        dic = {}
        for i in range(len(s)):

            if dic.get(s[i], 0) != 0 and dic.get(s[i]) != t[i]:
                return False
            elif t[i] in dic.values() and dic.get(s[i],0) != t[i]: return False
           
            dic[s[i]] = t[i]
            
        
        return True

        
